codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,237 @@ public TextHeaders clear() {
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence convertAndCommaSeparate(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    convertAndQuote(singleValue, values[i]).append(", ");
+                }
+                convertAndQuote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence commaSeparate(CharSequence[] values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    quote(singleValue, values[i]).append(", ");
+                }
+                quote(singleValue, values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence convertAndCommaSeparate(Iterable<?> values) {
+            StringBuilder sb = new StringBuilder();
+            Iterator<?> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                Object next = iterator.next();
+                while (iterator.hasNext()) {
+                    convertAndQuote(sb, next).append(", ");
+                    next = iterator.next();
+                }
+                convertAndQuote(sb, next);
+            }
+            return sb.toString();
+        }
+
+        private CharSequence commaSeparate(Iterable<? extends CharSequence> values) {
+            StringBuilder sb = new StringBuilder();
+            Iterator<? extends CharSequence> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                CharSequence next = iterator.next();
+                while (iterator.hasNext()) {
+                    quote(sb, next).append(", ");
+                    next = iterator.next();
+                }
+                quote(sb, next);
+            }
+            return sb.toString();
+        }
+
+        private StringBuilder convertAndQuote(StringBuilder sb, Object value) {
+            return quote(sb, valueConverter.convertObject(value));
+        }
+
+        private StringBuilder quote(StringBuilder sb, CharSequence value) {
+            String converted = value.toString();
+            if (converted.isEmpty()) {
+                return sb;
+            }
+            return converted.contains(COMMA) && converted.charAt(0) != DOUBLE_QUOTE ?
+                    sb.append(DOUBLE_QUOTE).append(converted).append(DOUBLE_QUOTE):
+                    sb.append(converted);
+        }
+
+        private CharSequence commaSeparate(CharSequence currentValue, CharSequence value) {
+            return new StringBuilder().append(currentValue).append(", ").append(value).toString();
```

Comment:
I think we should first try to consolidate `commaSeparate` and `convertAndCommaSeparate` as suggested in https://github.com/netty/netty/pull/3271/files#r22869589, but I think we may need to check for escaping `value`.